src/store/middleware.js

```diff
@@ -6,25 +6,45 @@ import {
   SET_MANUAL_LOCATION,
   SET_MAP_LOCATION,
   setProjectionResult,
+  SET_RECOMMENDATION_MODE,
 } from './actions';
 
 const projectionActionTypes = [
   SET_PROJECTION_MODE,
   SET_MANUAL_LOCATION,
   SET_MAP_LOCATION,
+  SET_RECOMMENDATION_MODE,
 ];
 
+function getRecommendationMode(recommendationMode, mapLocation) {
+  let recommendationModeText;
+  if (recommendationMode === 'today') {
+    recommendationModeText = mapLocation.altitudinalZone;
+  } else if (recommendationMode === 'moderate') {
+    recommendationModeText = mapLocation.targetAltitudinalZoneModerate;
+  } else if (recommendationMode === 'extreme') {
+    recommendationModeText = mapLocation.targetAltitudinalZoneExtreme;
+  }
+  return recommendationModeText;
+}
+
 export const projection = store => next => action => {
   const result = next(action);
   if (projectionActionTypes.includes(action.type)) {
-    const { manualLocation, mapLocation, projectionMode } = store.getState();
+    const {
+      manualLocation,
+      mapLocation,
+      projectionMode,
+      recommendationMode,
+    } = store.getState();
     const location =
       projectionMode === 'map'
         ? { ...manualLocation, ...mapLocation }
         : { ...mapLocation, ...manualLocation };
+
     const targetAltitudinalZone =
       projectionMode === 'map'
```

Comment:
Please refactor this into a function "getTargetAltitudinalZone" which accepts "location" and "recommendationMode".